src/adapter/src/catalog.rs

```diff
@@ -6662,6 +6663,12 @@ impl SessionCatalog for ConnCatalog<'_> {
     fn aws_privatelink_availability_zones(&self) -> Option<BTreeSet<String>> {
         self.state.aws_privatelink_availability_zones.clone()
     }
+
+    fn is_enabled(&self, feature_name: &str) -> bool {
+        let result = self.state.system_configuration.is_true(feature_name);
+        soft_assert!(result.is_ok());
```

Comment:
At the moment the `soft_assert` ensures that the `feature_name` value references an actual `SystemVar<bool>` field.
I can also change the signature of the enclosing method if there is a way to convert an `AdapterError` into a `SqlCatalogError`.